dlt/pipeline/pipeline.py

```diff
@@ -1103,6 +1108,10 @@ def _managed_state(self, *, extract_state: bool = False) -> Iterator[TPipelineSt
             # check if any state element was changed
             merged_state = merge_state_if_changed(backup_state, state)
 
+            # Force exctract when state hasn't changed
+            if extract_unchanged:
```

Comment:
why we need that? if there is nothing to sync then we can skip that part. do you see any problem with that?

Reply:
It's a bit of an wonky state to get the pipeline in though.  
Might be ok to always run sync destination at the beginning instead (after checking for pending data)